### PR TITLE
Dropdown menu aria enhancement

### LINKED ISSUE #3183: Enhance dropdown menu with aria attributes
We can enhance the dropdown menu component with additional aria attributes to inform assistive technology of its state e.g. whether it is open or closed. 

An article with some ideas on implementation is here: http://heydonworks.com/practical_aria_examples/#progressive-collapsibles

### PR DESCRIPTION
**Pull request by vritant24**
Fixes #3183 

Added `aria` and `role` attributes to the `DropdownMenu` component to inform assistive technology of the state and roles of the dropdown menu.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -150,6 +150,7 @@
   },
   "homepage": "https://github.com/mozillla/addons-frontend#readme",
   "dependencies": {
+    "all": "0.0.0",
```
Comment (willdurand) [inline package.json]: I suspect this is not needed. Could you remove it?

Maybe running `yarn remove all` would work. You should remove the changes in this file and in the `yarn.lock` file.

Comment (vritant24) [reply package.json]: Thanks for pointing that out. fixed it.

**Comment by vritant24** [comment]
So there is check failing in Travis CI, where `TOKENV=dennis-lint`
Could I get some help on how I would fix this?

**Comment by willdurand** [comment]
> Could I get some help on how I would fix this?

You can safely ignore this failure. It is not related to your changes.

Could you give some hints to test your ARIA changes? Like what these attributes are supposed to do for instance?

**Comment by vritant24** [comment]
Yes. Sorry for the late reply, it being holidays and such.
Here's where I referenced my implementation from http://heydonworks.com/practical_aria_examples/#submenus
It can be tested using assisstive technologies, and they will be able to identify that the dropdowns are indeed dropdowns, and have submenus.

**Comment by caitmuenster** [comment]
Thanks so much for this patch, @vritant24! It has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition#March_2018). 

Would you be interested in creating an account on [mozillians.org](https://mozillians.org)? I'd be happy to vouch for you!

**Comment by vritant24** [comment]
@tofumatt no problem! I'm glad I was able to contribute!

@caitmuenster I would be interested. I will go ahead and create an account.

**Comment by caitmuenster** [comment]
@vritant24 Awesome! Your profile is vouched. 

Welcome onboard! I look forward to seeing you around the project. :)

**Pull request merged**